src/Microsoft.Diagnostics.ExtensionCommands/ClrMDHelper.cs

```diff
@@ -19,7 +19,17 @@ public class ClrMDHelper
         [ServiceExport(Scope = ServiceScope.Runtime)]
         public static ClrMDHelper TryCreate([ServiceImport(Optional = true)] ClrRuntime clrRuntime)
         {
-            return clrRuntime != null ? new ClrMDHelper(clrRuntime) : null;
+            try
+            {
+                if (clrRuntime != null)
+                {
+                    return new ClrMDHelper(clrRuntime);
+                }
+            }
+            catch (NotSupportedException)
```

Comment:
Yes, CLRMD throws NotSupportedException when it can create the ClrRuntime.Heap.  It happens when the dump doesn't contain the heap or type system data structures. I think I ran into it in my testing on my Windows machine that didn't have the MiniDumpAuxiliaryDlls in registry set to the runtime I was testing. Not catching the exception here caused problems with the Optional=true attributed ClrMDHelper instances in commands or command filters.  

CLRMD doesn't give us any more detail about why.  See [ClrRuntime.cs](https://dnceng.visualstudio.com/internal/_git/Microsoft-clrmd?path=/src/Microsoft.Diagnostics.Runtime/ClrRuntime.cs&version=GBmain&line=215&lineEnd=216&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents).